fem/fespace.cpp

```diff
@@ -2428,6 +2428,7 @@ void FiniteElementSpace::Construct()
       {
          // the simple case: all edges are of the same order
          nedofs = mesh->GetNEdges() * fec->GetNumDof(Geometry::SEGMENT, order);
+         var_edge_dofs.Clear(); // ensure any old var_edge_dof table is dumped.
```

Comment:
This works for the `ex15p` test in the PR description, but I wonder if this line should be after the if-else, so that it is done for both cases. The definition of this table has a comment saying it is used only in the variable-order case, and it is empty otherwise. The fact that it needs to be cleared here implies it is not empty in the constant-order case, right? If so, the comment in the header should be updated. The same goes for `var_face_dofs`.

Reply:
`var_face_dofs` is used for mixed meshes in the constant case. Clearing `var_edge_dofs` here I did for a) symmetry and b) safety. If the space has switched to all same order (for instance p refinement resulted in all local elements having the same p, after having previously had different p), then it would store invalid data from the previous mesh. So clearing it like for `var_face_dofs` seemed prudent, as that caused definite errors.

The issue is that for a previous mesh, `var_edge_dofs` might have been non-empty, but if the space then entered this code path, it would now hold invalid data about the previous space.

Reply:
Thanks @hughcars, this makes sense. I see that in the variable-order case, it clears and reconstructs these tables, whereas the constant-order case just needs a clearing.